app/src/main/java/com/aconno/sensorics/ui/device_main/DeviceMainFragment.kt

```diff
@@ -162,10 +162,7 @@ class DeviceMainFragment : DaggerFragment() {
     override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
         super.onCreateOptionsMenu(menu, inflater)
         activity?.menuInflater?.inflate(R.menu.menu_readings, menu)
-        if (mDevice.connectable) {
-            menu.clear()
-            this.menu = menu
-        }
+        this.menu = menu
```

Comment:
Are you sure `menu.clear()` is not needed?

Reply:
Clearing all menu items just doesn't make any sense, especially because that means that you don't have connect/disconnect button which is needed if you want to connect or disconnect to a connectable device (`menu.clear()` is called when the device is connectable).

Reply:
Look at [example usages](https://www.codota.com/code/java/methods/android.view.Menu/clear) of `menu.clear()`.

Most of the time it is used when we want to clear the menu items added by different fragments. I think we have a bug where we first inflate our menu, and then we clear it. This results in empty menu. We should first clear it, and then inflate it. What do you think?

We should not just delete this code. Did you test it thoroughly, and verify that we don't have unwanted menu items?

Reply:
But menu.clear() is called inside the following if statement:
```
if (mDevice.connectable) {
            menu.clear()
            this.menu = menu
        }
```
So the intention of author of this code was clearly not to clear the menu items added by different fragments, because if it was, then it would not be called only if device is connectable.

It is hard to tell if some menu items are unwanted. There are the following menu items shown that are not directly correlated to the device: Actions, App Settings, Buy beacons and About